src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ClientSkinSystem.java

```diff
@@ -55,24 +57,54 @@ public void initialise() {
+     * Default event handler for this event has Trivial priority. Hence, this method catches the event first
+     * and consumes it.
+     * @see CreateVisualCharacterEvent
+     * @see org.terasology.logic.characters.VisualCharacterSystem
      *
      * @param event
-     * @param entity
+     * @param characterEntity
+     * @param lasTeamComponent
      */
     @ReceiveEvent
-    public void onAddPlayerSkinToPlayer(AddPlayerSkinToPlayerEvent event, EntityRef entity) {
-        EntityRef player = event.player;
-        String team = event.team;
-        if (team.equals(LASUtils.BLACK_TEAM)) {
-            builder = entityManager.newBuilder(LASUtils.BLACK_PAWN);
-            builder.saveComponent(player.getComponent(LocationComponent.class));
-            builder.build();
+    public void onCreateDefaultVisualCharacter(CreateVisualCharacterEvent event, EntityRef characterEntity,
+                                               LASTeamComponent lasTeamComponent) {
+        Prefab prefab = assetManager.getAsset("engine:defaultVisualCharacter", Prefab.class).get();
+        EntityBuilder entityBuilder = event.getVisualCharacterBuilder();
+        entityBuilder.addPrefab(prefab);
+        SkeletalMeshComponent skeletalMeshComponent = entityBuilder.getComponent(SkeletalMeshComponent.class);
+        if (lasTeamComponent.team.equals(LASUtils.BLACK_TEAM)) {
+            skeletalMeshComponent.material = Assets.getMaterial(LASUtils.BLACK_PAWN_SKIN).get();
+        } else if (lasTeamComponent.team.equals(LASUtils.RED_TEAM)) {
+            skeletalMeshComponent.material = Assets.getMaterial(LASUtils.RED_PAWN_SKIN).get();
         }
-        if (team.equals(LASUtils.RED_TEAM)) {
-            builder = entityManager.newBuilder(LASUtils.RED_PAWN);
-            builder.saveComponent(player.getComponent(LocationComponent.class));
-            builder.build();
+        entityBuilder.saveComponent(skeletalMeshComponent);
+        event.consume();
+    }
+
+    /**
+     * Updates the skeletal mesh of a player when its team changes.
+     * @see LASTeamComponent
+     *
+     * @param event
+     * @param characterEntity
+     * @param lasTeamComponent
+     */
+    @ReceiveEvent
+    public void onLASTeamChange(OnChangedComponent event, EntityRef characterEntity, LASTeamComponent lasTeamComponent) {
+        if (characterEntity.hasComponent(VisualCharacterComponent.class)) {
+            VisualCharacterComponent visualCharacterComponent = characterEntity.getComponent(VisualCharacterComponent.class);
+            EntityRef visualCharacter = visualCharacterComponent.visualCharacter;
+            if (visualCharacter != EntityRef.NULL && visualCharacter.hasComponent(SkeletalMeshComponent.class)) {
+                SkeletalMeshComponent skeletalMeshComponent = visualCharacter.getComponent(SkeletalMeshComponent.class);
+                if (lasTeamComponent.team.equals(LASUtils.BLACK_TEAM)) {
```

Comment:
Instead of having the logic below for team check and skeletal mesh update, can you do something similar as done in the case of Health HUD? i.e. create a method in LASUtils to give the respective Pawn skin after checking the team, as in-

https://github.com/Terasology/LightAndShadow/blob/3d894f46c2c32af8e33db0e187c997313e8aab4d/src/main/java/org/terasology/ligthandshadow/componentsystem/LASUtils.java#L117-L127

Reply:
Done!